### PR TITLE
test: refactors MainPresenter and adds unit tests

### PR DESCRIPTION
**Pull request by ferdyrod**

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/main/MainPresenter.kt [hunk 1]
```diff
@@ -20,60 +21,55 @@ const val DEFAULT = "default"
...
         )
 
-        compositeDisposable!!.add(
-            d2.categoryModule().categoryCombos().byIsDefault().eq(true).one().get().toObservable()
+        disposable.add(
+            d2.categoryModule().categoryCombos().byIsDefault().eq(true).one().get()
                 .subscribeOn(schedulerProvider.io())
                 .subscribe(
                     { categoryCombo ->
-                        val prefs = view.abstracContext.getSharedPreferences(
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/main/MainPresenter.kt]: :+1:

File: app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    private val preferences: PreferenceProvider = mock()
+    private val workManger: WorkManager = mock()
+
+    @Before
+    fun setUp() {
+        presenter = MainPresenter(view, d2, schedulers, preferences, workManger)
+    }
+
+    @Test
+    fun `Should show username on activity resumed`() {
```
Comment (JaimeToca) [inline app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt]: These tests belong to the same **flow**, function and usecase which is init() the presener. Perhaps they should be compacted as one test... Let me know what you think :+1: 

`presenterMocks()
presenter.init()
//all verifies`

File: app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    @Test
+    fun `Should save default categoryOptionCombo uid to preferences`() {
+        presenterMocks()
+
+        presenter.init()
+
+        verify(preferences).setValue(PREF_DEFAULT_CAT_OPTION_COMBO, "uid")
+    }
```
Comment (JaimeToca) [inline app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt]: These 4 tests belong to the same flow, function and usecase which is init() the presener. Perhaps they should be compacted as one test... Let me know what you think +1

presenterMocks() 
presenter.init() //all verifies

File: app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    @Test
+    fun `Should open drawer when menu is clicked`() {
+        presenter.onMenuClick()
+
+        verify(view).openDrawer(any())
+    }
+
+    private fun presenterMocks() {
+        // UserModule
+        whenever(d2.userModule()) doReturn mock()
```
Comment (JaimeToca) [inline app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt]: I know all this mocking is huge. We'll try to return mocks from sdk dagger modules... I'll keep you updated ;)

**Comment by JaimeToca** [comment]
Very nice PR @ferdyrod ! I posted a few comments. Let me know what you think ;)

cheers

File: app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt (hunk 3 above)
Comment (ferdyrod) [reply app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt]: I completely agree with this.

**Pull request merged**